Src/BufferHelpers.cpp

```diff
@@ -0,0 +1,281 @@
+//------------------------------------- -------------------------------------------------
+// File: BufferHelpers.cpp
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+//
+// http://go.microsoft.com/fwlink/?LinkID=615561
+//--------------------------------------------------------------------------------------
+
+#include "pch.h"
+#include "BufferHelpers.h"
+#include "DirectXHelpers.h"
+#include "ResourceUploadBatch.h"
+#include "LoaderHelpers.h"
+#include "PlatformHelpers.h"
+
+
+using namespace DirectX;
+using Microsoft::WRL::ComPtr;
+
+//--------------------------------------------------------------------------------------
+_Use_decl_annotations_
+HRESULT DirectX::CreateStaticBuffer(
+    ID3D12Device* device,
+    ResourceUploadBatch& resourceUpload,
+    const void* ptr,
+    size_t count,
+    size_t stride,
+    D3D12_RESOURCE_STATES afterState,
+    ID3D12Resource** pBuffer) noexcept
+{
+    if (!pBuffer)
+        return E_INVALIDARG;
+
+    *pBuffer = nullptr;
+
+    if (!device || !ptr || !count || !stride)
+        return E_INVALIDARG;
+
+    uint64_t bytes = uint64_t(count) * uint64_t(stride);
+    if (bytes >= UINT32_MAX)
+        return HRESULT_FROM_WIN32(ERROR_ARITHMETIC_OVERFLOW);
+
+    auto desc = CD3DX12_RESOURCE_DESC::Buffer(bytes);
+
+    CD3DX12_HEAP_PROPERTIES heapProperties(D3D12_HEAP_TYPE_DEFAULT);
+
+    ComPtr<ID3D12Resource> res;
+    HRESULT hr = device->CreateCommittedResource(
+        &heapProperties,
+        D3D12_HEAP_FLAG_NONE,
+        &desc,
+        D3D12_RESOURCE_STATE_COPY_DEST,
+        nullptr,
+        IID_GRAPHICS_PPV_ARGS(res.GetAddressOf()));
+    if (FAILED(hr))
+        return hr;
+
+    D3D12_SUBRESOURCE_DATA initData = { ptr, static_cast<LONG_PTR>(bytes), 0 };
+
+    try
+    {
+        resourceUpload.Upload(res.Get(), 0, &initData, 1);
+
+        resourceUpload.Transition(res.Get(), D3D12_RESOURCE_STATE_COPY_DEST, afterState);
+    }
+    catch (com_exception e)
+    {
+        return e.get_result();
+    }
+    catch (...)
+    {
+        return E_FAIL;
+    }
+
+    *pBuffer = res.Detach();
+
+    return S_OK;
+}
+
+
+//--------------------------------------------------------------------------------------
+_Use_decl_annotations_
+HRESULT DirectX::CreateTextureFromMemory(
+    ID3D12Device* device,
+    ResourceUploadBatch& resourceUpload,
+    size_t width,
```

Comment:
Should this just be a uint32_t?  I noticed in other places you changed from size_t to uint32_t... so maybe it makes sense to do that here, too?